tavern_server/src/db_tables.sql

```diff
@@ -0,0 +1,861 @@
+	'Climb', 
+	'Craft', 
+	'Diplomacy', 
+	'Disable Device', 
+	'Disguise', 
+	'Escape Artist', 
+	'Fly', 
+	'Handle Animal', 
+	'Heal', 
+	'Intimidate', 
+	'Knowledge (arcana)', 
+	'Knowledge (dungeoneering)', 
+	'Knowledge (engineering)', 
+	'Knowledge (geography)', 
+	'Knowledge (history)', 
+	'Knowledge (local)', 
+	'Knowledge (nobility)', 
+	'Knowledge (planes)', 
+	'Knowledge (religion)', 
+	'Linguistics', 
+	'Perception', 
+	'Perform', 
+	'Profession', 
+	'Ride', 
+	'Sense Motive', 
+	'Sleight of Hand', 
+	'Spellcraft', 
+	'Stealth', 
+	'Survival', 
+	'Swim', 
+	'Use Magic Device'
+);
+
+CREATE TYPE IF NOT EXISTS SaveThrow AS ENUM (
+	'Fortitude', 
+	'Reflex',
+	'Will'
+);
+
+CREATE TYPE IF NOT EXISTS CharacterStat AS ENUM (
+	'Name',
+	'Race',
+	'Size',
+	'Height',
+	'Weight',
+	'Age',
+	'Gender',
+	'Alignment',
+	'Deity',
+	'Languages',
+	'Appearance'
+);
+
+CREATE TYPE IF NOT EXISTS CombatStat AS ENUM (
+	'Melee Attack Bonus',
+	'Ranged Attack Bonus',
+	'CMB',
+	'CMD',
+	'Armor Class',
+	'Touch AC',
+	'Flat-Footed AC',
+	'Initiative Bonus',
+	'Damage Reduction',
+	'Spell Resistance',
+	'Speed',
+	'Fortitude Save',
+	'Reflex Save',
+	'Will Save'
+);
+
+CREATE TYPE IF NOT EXISTS RaceType AS ENUM (
+	'Abberation', 
+	'Animal', 
+	'Construct', 
+	'Dragon', 
+	'Fey', 
+	'Humanoid', 
+	'Magical Beast', 
+	'Monstrous Humanoid', 
+	'Ooze', 
+	'Outsider', 
+	'Plant', 
+	'Undead', 
+	'Vermin'
+);
+
+CREATE TYPE IF NOT EXISTS RaceSubtyep AS ENUM (
+
+);
+
+CREATE TYPE IF NOT EXISTS CasterType AS ENUM (
+	'Spontaneous', 
+	'Prepared'
+);
+
+CREATE TYPE IF NOT EXISTS ComponentType AS ENUM (
+	'Somatic',
+	'Material',
+	'Verbal'
+);
+
+CREATE TYPE IF NOT EXISTS MagicSchool AS ENUM (
+	'Abjuration', 
+	'Conjuration', 
+	'Divination', 
+	'Enchantment', 
+	'Evocation', 
+	'Illusion', 
+	'Necromancy', 
+	'Transmutation'
+);
+
+CREATE TYPE IF NOT EXISTS SpellRange AS ENUM (
+	'Personal',
+	'Touch',
+	'Close',
+	'Medium',
+	'Long',
+	'Unlimited'
+);
+
+CREATE TYPE IF NOT EXISTS WeaponClass AS ENUM (
+	'Axes', 
+	'Heavy Blades', 
+	'Light Blades', 
+	'Bows', 
+	'Close', 
+	'Crossbows', 
+	'Double', 
+	'Firearms', 
+	'Flails', 
+	'Hammers', 
+	'Monk', 
+	'Natural', 
+	'Polearms', 
+	'Siege Engines', 
+	'Spears', 
+	'Thrown', 
+	'Tribal'
+);
+
+CREATE TYPE IF NOT EXISTS ArmorClass AS ENUM (
+	'Light', 
+	'Medium', 
+	'Heavy'
+);
+
+CREATE TYPE IF NOT EXISTS EquipmentSlot AS ENUM (
+	'None',
+	'Armor', 
+	'Belts', 
+	'Body', 
+	'Chest', 
+	'Eyes', 
+	'Feet', 
+	'Hands', 
+	'Head', 
+	'Headband', 
+	'Neck', 
+	'Ring (left)', 
+	'Ring (right)', 
+	'Shield', 
+	'Shoulders', 
+	'Wrist'
+);
+
+CREATE TYPE IF NOT EXISTS DamageType AS ENUM (
+	'Bludgeoning', 
+	'Slashing', 
+	'Piercing', 
+	'Energy', 
+	'Acid', 
+	'Fire', 
+	'Electricity', 
+	'Cold', 
+	'Sonic', 
+	'Positive', 
+	'Negative', 
+	'Nonlethal'
+);
+
+--table declarations
+--user creds table - also used for hashing info
+CREATE TABLE IF NOT EXISTS Users (
+	id     			UUID PRIMARY KEY,
+
+	email			TEXT 
```

Comment:
Should we include a check here to ensure the email address fits a pattern? It seems possible in Postgres.

Reply:
I can look into this , but also "email regex" sends shivers down my spine.

Reply:
As it [well should](https://www.regular-expressions.info/email.html). I was thinking, if we did any checking, it would be a general `[thing]@[thing].[thing]`, so `a@b.c` would be a valid email address. Any extra validation would come later in the form of sending a confirmation email before the account is fully created.